### ISSUE TITLE
Use the new build system

### ISSUE THREAD

**Issue by kevmoo** [dart-lang/intl_translation#32]
Would likely be very helpful with frequent builds...

**Comment by enyo** [comment]
Yes! The first step towards this, would be to remove most of the logic from `bin/generate_from_arb.dart` and put it inside `lib/`. Then at least we would be able to write our own builders for it.
Right now, there is a lot of logic inside the `bin/generate_from_arb.dart` file that I need to copy into my builder to make it work.

**Comment by dotdoom** [comment]
Clarification: new build system is [`package:build`](https://pub.dartlang.org/packages/build).

**Comment by kevmoo** [comment]
...and friend :-)

**Comment by alan-knight** [comment]
Hmm. I wonder if it would actually be preferable to remove it to a separate package entirely. The general logic is already in generate_localized.dart, and the idea is that readers/writers for a particular format only need format-specific information. Having that in a separate package would make it easier for people to use, though at the cost of it being harder to use a default.

**Comment by kevmoo** [comment]
We should look at having a mono-repo if we create yet another intl package.

Keeping these in one repo would be nice...

**Comment by dotdoom** [comment]
Pardon my ignorance, but if `analyzer` is part of `dart-sdk` anyway, what's the point of this package, `intl_translation`? It seems trivial to have `intl` depend on `analyzer`.

**Comment by alan-knight** [comment]
Analyzer ships with the dart sdk but is not part of it. That is, a Dart application that depends on the SDK does not depend on on the analyzer. If an application uses Intl at run-time, having it depend on analyzer would increase the code size significantly.

**Comment by dotdoom** [comment]
I was kind of expecting that Dart's tree shaking would remove `analyzer` if it's not being used (even though it's listed in dependencies).

But even then, wouldn't it be possible to have `analyzer` in `dev_dependencies`?
